pkg/bootstrap/bootstrap_config.go

```diff
@@ -178,8 +178,10 @@ func extractMetadata(envs []string, prefix string, set setMetaFunc, meta map[str
 // ISTIO_METAJSON_* env variables contain json_string in the value.
 // 					The name of variable is ignored.
 // ISTIO_META_* env variables are passed thru
-func getNodeMetaData(envs []string) map[string]string {
-	meta := map[string]string{}
+func getNodeMetaData(envs []string, meta map[string]string) map[string]string {
+	if meta == nil {
```

Comment:
This is likely overly defensive, but I wonder if we should treat `config.EnvoyNodeMetadata` as immutable and copy its values over rather than just mutating the original config object. The name `getNodeMetaData` doesn't really imply that it would mutate its arguments.

Reply:
I agree that the function name is not great. I just changed it.